src/hooks/__internal__/useModalAria/useModalAria.spec.tsx

```diff
@@ -0,0 +1,305 @@
+import React, { useRef, useState } from "react";
+import { render, screen, waitFor } from "@testing-library/react";
+import userEvent from "@testing-library/user-event";
+import useModalManager from "../useModalManager";
+import useModalAria from ".";
+import CarbonProvider from "../../../components/carbon-provider/carbon-provider.component";
+
+interface ModalComponentProps {
+  openButtonText: string;
+  closeButtonText: string;
+  dialogLabel?: string;
+  children?: React.ReactNode;
+}
+
+const FakeModal = ({
```

Comment:
**nitpick/opinion:** Super nitpicky comment, but i'm not 100% sure i'm a fan of the name `FakeModel`, which is quite mad considering i'm fairly well known for my awful naming conventions. Would `MockModal` be a better name, perhaps? Other synonyms which are more descriptive of the purpose could be: `InertModal`, `SyntheticModal`, `ArtificialModal` etc. But, happy for you to stick with `FakeModal`, if you don't think any of these / any other names are any better 👍

Reply:
fair comment, I didn't really think about it too much - I agree MockModal would probably be more in keeping with what we do elsewhere. I'll wait to see what the other reviewer says but more than happy to change 👍 

For what it's worth, I was originally using an actual Carbon `Modal` component here, but that caused issues in the tests for resetting the properties of outside containers correctly, which to cut a long story short were due to Modal rendering the content inside a Portal. So I decided to make my own "fake modal" which doesn't use a portal - but it has to call `useModalManager` to manage the tracking of the top modal correctly so that's how we ended up here 😅 But yeah, the name could probably use some work!